test/test_jit.py

```diff
@@ -2061,6 +2061,147 @@ def forward(self, inp):
         m2 = M2()
         m2(torch.zeros(4, 3))
 
+    def test_script_module_const(self):
+        class M(torch.jit.ScriptModule):
+
+            __constants__ = ['b', 'i', 'c']
+
+            def __init__(self):
+                super(M, self).__init__(False)
+                self.b = False
+                self.i = 1
+                self.c = 3.5
+
+            @torch.jit.script_method
+            def forward(self):
+                return self.b, self.i, self.c
+
+        m = M()
+        o0, o1, o2 = m()
+        self.assertEqual(o0, 0)
+        self.assertEqual(o1, 1)
+        self.assertEqual(o2, 3.5)
```

Comment:
Will those be actual Python numbers or PyTorch scalars?

Reply:
They are PyTorch scalars, I think the assertEqual is just smart enough to do the comparison.